src_c/display.c

```diff
@@ -1607,6 +1608,11 @@ pg_list_modes(PyObject *self, PyObject *args, PyObject *kwds)
             Py_DECREF(list);
             return NULL;
         }
+        if (last_width == mode.w && last_height == mode.h && last_width != -1) {
+            continue;
```

Comment:
Leaks `size` here.  My recommendation is to move this bit up before we create the `size` tuple to side step the issue.

Alternatively, we need a `Py_DECREF(size);` before the `continue;`

Reply:
> move this bit up before we create the size tuple to side step the issue
Indeed, this would fix it.

Reply:
Thank you @nthykier . Can you check it now to see if it is ok